java-checks/src/main/java/org/sonar/java/checks/security/IntegerToHexStringCheck.java

```diff
@@ -0,0 +1,90 @@
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.java.checks.security;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.Optional;
+import org.sonar.check.Rule;
+import org.sonar.java.checks.methods.AbstractMethodDetection;
+import org.sonar.java.matcher.MethodMatcher;
+import org.sonar.java.matcher.TypeCriteria;
+import org.sonar.plugins.java.api.tree.BinaryExpressionTree;
+import org.sonar.plugins.java.api.tree.ExpressionTree;
+import org.sonar.plugins.java.api.tree.LiteralTree;
+import org.sonar.plugins.java.api.tree.MethodInvocationTree;
+import org.sonar.plugins.java.api.tree.Tree;
+
+@Rule(key = "S4425")
+public class IntegerToHexStringCheck extends AbstractMethodDetection {
+
+  private static final MethodMatcher APPEND_MATCHER = MethodMatcher.create()
+    .typeDefinition(TypeCriteria.subtypeOf("java.lang.AbstractStringBuilder"))
+    .name("append")
+    .addParameter(TypeCriteria.is("java.lang.String"));
+
+  private static final MethodMatcher PRINT_MATCHER = MethodMatcher.create()
+    .typeDefinition(TypeCriteria.subtypeOf("java.io.PrintStream"))
+    .name("print")
+    .addParameter(TypeCriteria.is("java.lang.String"));
+
+  @Override
+  protected List<MethodMatcher> getMethodInvocationMatchers() {
+    return Collections.singletonList(
+      MethodMatcher.create()
+        .typeDefinition(TypeCriteria.is("java.lang.Integer"))
+        .name("toHexString")
+        .addParameter(TypeCriteria.is("int")));
+  }
+
+  @Override
+  protected void onMethodInvocationFound(MethodInvocationTree method) {
+    if (isArgumentAppended(method) && typeIsByte(method.arguments().get(0))) {
+      reportIssue(method.methodSelect(), "Use String.format( \"%02X\", ...) instead.");
+    }
+  }
+
+  private static boolean isArgumentAppended(MethodInvocationTree method) {
+    return Optional.of(method)
+      .map(Tree::parent)
+      .filter(tree -> tree.is(Tree.Kind.ARGUMENTS))
+      .map(Tree::parent)
+      .filter(tree -> tree.is(Tree.Kind.METHOD_INVOCATION))
+      .map(MethodInvocationTree.class::cast)
+      .filter(parentMethod-> APPEND_MATCHER.matches(parentMethod) || PRINT_MATCHER.matches(parentMethod))
+      .isPresent();
+  }
+
+  private static boolean typeIsByte(ExpressionTree expression) {
+    if (expression.symbolType().isSubtypeOf("byte")) {
+      return true;
+    }
+    if (expression.is(Tree.Kind.AND)) {
+      BinaryExpressionTree and = (BinaryExpressionTree) expression;
```

Comment:
This appears in `RawByteBitwiseOperationsCheck.isSecuringByte` method, could you move both to a common method in `LiteralUtils`?